README.md
# sendcode README

This is the README for your extension "sendcode". After writing up a brief description, we recommend including the following sections.

## Features

Describe specific features of your extension including screenshots of your extension in action. Image paths are relative to this README file.

For example if there is an image subfolder under your extension project workspace:

\!\[feature X\]\(images/feature-x.png\)

> Tip: Many popular extensions utilize animations. This is an excellent way to show off your extension! We recommend short, focused animations that are easy to follow.

## Requirements

If you have any requirements or dependencies, add a section describing those and how to install and configure them.

## Extension Settings

Include if your extension adds any VS Code settings through the `contributes.configuration` extension point.

For example:

This extension contributes the following settings:

* `myExtension.enable`: Enable/disable this extension.
* `myExtension.thing`: Set to `blah` to do something.

## Known Issues

Calling out known issues can help limit users opening duplicate issues against your extension.

## Release Notes

Users appreciate release notes as you update your extension.

### 1.0.0

Initial release of ...

### 1.0.1

Fixed issue #.

### 1.1.0

Added features X, Y, and Z.

---

## Following extension guidelines

Ensure that you've read through the extensions guidelines and follow the best practices for creating your extension.

* [Extension Guidelines](https://code.visualstudio.com/api/references/extension-guidelines)

## Working with Markdown

You can author your README using Visual Studio Code. Here are some useful editor keyboard shortcuts:

* Split the editor (`Cmd+\` on macOS or `Ctrl+\` on Windows and Linux).
* Toggle preview (`Shift+Cmd+V` on macOS or `Shift+Ctrl+V` on Windows and Linux).
* Press `Ctrl+Space` (Windows, Linux, macOS) to see a list of Markdown snippets.

## For more information

* [Visual Studio Code's Markdown Support](http://code.visualstudio.com/docs/languages/markdown)
* [Markdown Syntax Reference](https://help.github.com/articles/markdown-basics/)

**Enjoy!**
# sendData-


CHANGELOG.md
# Change Log

All notable changes to the "sendcode" extension will be documented in this file.

Check [Keep a Changelog](http://keepachangelog.com/) for recommendations on how to structure this file.

## [Unreleased]

- Initial release

out/extension.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.activate = void 0;
const openai_1 = require("openai");
const fs = require("fs");
const path = require("path");
class ChatGPTClient {
    constructor() {
        const configuration = new openai_1.Configuration({
            apiKey: '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',
        });
        this.openAI = new openai_1.OpenAIApi(configuration);
    }
    async respond(chatGPTMessages) {
        try {
            if (!chatGPTMessages) {
                return {
                    text: 'No chatGPT Messages',
                };
            }
            const request = {
                messages: chatGPTMessages,
                model: 'gpt-3.5-turbo',
            };
            const response = await this.openAI.createChatCompletion(request);
            if (!response.data || !response.data.choices) {
                return {
                    text: 'The bot did not respond. Please try again later.',
                };
            }
            return {
                text: response.data.choices[0].message?.content,
                messageId: response.data.id,
            };
        }
        catch (error) {
            console.log(error);
            throw new Error(String(error));
        }
    }
}
// The module 'vscode' contains the VS Code extensibility API
// Import the module and reference it with the alias vscode in your code below
const vscode = require("vscode");
// This method is called when your extension is activated
// Your extension is activated the very first time the command is executed
function activate(context) {
    console.log('Congratulations, your extension is now active!');
    let disposable = vscode.commands.registerCommand('bunyan.generate', async () => {
        const editor = vscode.window.activeTextEditor;
        if (editor) {
            const document = editor.document;
            const originalUri = document.uri;
            const text = document.getText();
            // read text from style.txt
            let styleText;
            try {
                styleText = fs.readFileSync(path.resolve(__dirname, 'style.spec'), 'utf8');
            }
            catch (error) {
                console.error(`Error reading style.txt: ${error}`);
                return;
            }
            let newText = '';
            // Start progress
            await vscode.window.withProgress({
                location: vscode.ProgressLocation.Notification,
                title: "Processing your request",
                cancellable: false
            }, async (progress, token) => {
                progress.report({ increment: 0 });
                const chatGPTClient = new ChatGPTClient();
                const chatGPTMessage = {
                    role: 'user',
                    content: text,
                };
                const instructionMessage = {
                    role: 'assistant',
                    content: 'Make changes only when necessary. Respond with only code, no natural language.'
                };
                const chatGPTMessage2 = {
                    role: 'user',
                    content: `Given contents of a code and the style guide file, return the revised code following style guide: ${styleText}. Only code response`
                };
                const chatGPTRequest = {
                    model: 'gpt-4-32k',
                    messages: [instructionMessage, chatGPTMessage, chatGPTMessage2],
                };
                progress.report({ increment: 25 });
                await chatGPTClient
                    .respond([chatGPTMessage])
                    .then(response => {
                    console.log("response", response);
                    if (!response.text) {
                        newText = 'The bot did not respond. Please try again later.';
                        return;
                    }
                    newText = response.text;
                })
                    .catch(error => {
                    console.log(error);
                    newText = 'The bot did not respond. Please try again later.';
                });
                progress.report({ increment: 50 });
                // editor.edit(editBuilder => {
                //     const range = new vscode.Range(
                //         document.positionAt(0),
                //         document.positionAt(text.length)
                //     );
                //     editBuilder.replace(range, newText);
                // });
                // Create a new text document
                const newDocument = await vscode.workspace.openTextDocument({ content: newText });
                // Show the diff view
                vscode.commands.executeCommand('vscode.diff', document.uri, newDocument.uri, 'Original vs. Modified');
                progress.report({ increment: 75 });
                // Show information message with two options
                const result = await vscode.window.showInformationMessage('Do you want to apply these changes to the original file?', 'Yes', 'No');
                // If the user selects 'Yes', update the original document
                // If the user selects 'Yes', update the original document
                if (result === 'Yes') {
                    const originalDocument = await vscode.workspace.openTextDocument(originalUri); // Open the original document
                    const originalEditor = await vscode.window.showTextDocument(originalDocument); // Show the original document
                    // Apply the changes to the original document
                    await originalEditor.edit(editBuilder => {
                        const range = new vscode.Range(originalDocument.positionAt(0), originalDocument.positionAt(text.length));
                        editBuilder.replace(range, newText);
                    });
                }
                progress.report({ increment: 100 });
            });
        }
    });
}
exports.activate = activate;
//# sourceMappingURL=extension.js.map

src/extension.ts
import {
    OpenAIApi,
    Configuration,
    CreateChatCompletionRequest,
    ChatCompletionRequestMessage,
    ChatCompletionRequestMessageRoleEnum
} from 'openai';

import * as fs from 'fs';
import * as path from 'path';

class ChatGPTClient {
    private openAI: OpenAIApi;
    constructor() {
        const configuration = new Configuration({
            apiKey: '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',
        });
        this.openAI = new OpenAIApi(configuration);
    }
    async respond(chatGPTMessages: Array<ChatCompletionRequestMessage>) {
        try {
            if (!chatGPTMessages) {
                return {
                    text: 'No chatGPT Messages',
                };
            }
            const request: CreateChatCompletionRequest = {
                messages: chatGPTMessages,
                model: 'gpt-3.5-turbo',
            };
            const response = await this.openAI.createChatCompletion(request);
            if (!response.data || !response.data.choices) {
                return {
                    text: 'The bot did not respond. Please try again later.',
                };
            }
            return {
                text: response.data.choices[0].message?.content,
                messageId: response.data.id,
            };
        } catch (error) {
            console.log(error);
            throw new Error(String(error));
        }
    }
}

// The module 'vscode' contains the VS Code extensibility API
// Import the module and reference it with the alias vscode in your code below
import * as vscode from 'vscode';

// This method is called when your extension is activated
// Your extension is activated the very first time the command is executed
export function activate(context: vscode.ExtensionContext) {
    console.log('Congratulations, your extension is now active!');

    let disposable = vscode.commands.registerCommand('bunyan.generate', async () => {
        const editor = vscode.window.activeTextEditor;

        if (editor) {
            const document = editor.document;
            const originalUri = document.uri;
            const text = document.getText();
            
            // read text from style.txt
            let styleText: string;
            try {
                styleText = fs.readFileSync(path.resolve(__dirname, 'style.spec'), 'utf8');
            } catch (error) {
                console.error(`Error reading style.txt: ${error}`);
                return;
            }

            let newText = '';

            // Start progress
            await vscode.window.withProgress({
                location: vscode.ProgressLocation.Notification,
                title: "Processing your request",
                cancellable: false
            }, async (progress, token) => {
                progress.report({ increment: 0 });

                const chatGPTClient = new ChatGPTClient();

                const chatGPTMessage: ChatCompletionRequestMessage = {
                    role: 'user',
                    content: text,
                };

                const instructionMessage: ChatCompletionRequestMessage = {
                    role: 'assistant',
                    content: 'Make changes only when necessary. Respond with only code, no natural language.'
                };

                const chatGPTMessage2: ChatCompletionRequestMessage = {
                    role: 'user',
                    content: `Given contents of a code and the style guide file, return the revised code following style guide: ${styleText}. Only code response`
                };

                const chatGPTRequest: CreateChatCompletionRequest = {
                    model: 'gpt-4-32k',
                    messages: [instructionMessage, chatGPTMessage, chatGPTMessage2],
                };

                progress.report({ increment: 25 });

                await chatGPTClient
                    .respond([chatGPTMessage])
                    .then(response => {
                        console.log("response", response);
                        if (!response.text) {
                            newText = 'The bot did not respond. Please try again later.';
                            return;
                        }
                        newText = response.text;
                    })
                    .catch(error => {
                        console.log(error);
                        newText = 'The bot did not respond. Please try again later.';
                    });

                progress.report({ increment: 50 });

                // editor.edit(editBuilder => {
                //     const range = new vscode.Range(
                //         document.positionAt(0),
                //         document.positionAt(text.length)
                //     );
                //     editBuilder.replace(range, newText);
                // });

                // Create a new text document
                const newDocument = await vscode.workspace.openTextDocument({ content: newText });

                // Show the diff view
                vscode.commands.executeCommand('vscode.diff', document.uri, newDocument.uri, 'Original vs. Modified');

                progress.report({ increment: 75 });

                // Show information message with two options
                const result = await vscode.window.showInformationMessage('Do you want to apply these changes to the original file?', 'Yes', 'No');

                // If the user selects 'Yes', update the original document
                // If the user selects 'Yes', update the original document
                if (result === 'Yes') {
                    const originalDocument = await vscode.workspace.openTextDocument(originalUri); // Open the original document
                    const originalEditor = await vscode.window.showTextDocument(originalDocument); // Show the original document

                    // Apply the changes to the original document
                    await originalEditor.edit(editBuilder => {
                        const range = new vscode.Range(
                            originalDocument.positionAt(0),
                            originalDocument.positionAt(text.length)
                        );
                        editBuilder.replace(range, newText);
                    });
                }
                progress.report({ increment: 100 });
            });
        }
    });
}
